pkg/promlib/querydata/request.go

```diff
@@ -129,38 +152,51 @@ func (s *QueryData) fetch(traceCtx context.Context, client *client.Client, q *mo
 		Error:  nil,
 	}
 
+	var (
+		wg sync.WaitGroup
+		m  sync.Mutex
+	)
+
 	if q.InstantQuery {
-		res := s.instantQuery(traceCtx, client, q, enablePrometheusDataplane)
-		dr.Error = res.Error
-		dr.Frames = res.Frames
-		dr.Status = res.Status
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			res := s.instantQuery(traceCtx, client, q, enablePrometheusDataplane)
+			m.Lock()
+			addDataResponse(&res, dr)
+			m.Unlock()
+		}()
 	}
 
 	if q.RangeQuery {
-		res := s.rangeQuery(traceCtx, client, q, enablePrometheusDataplane)
-		if res.Error != nil {
-			if dr.Error == nil {
-				dr.Error = res.Error
-			} else {
-				dr.Error = fmt.Errorf("%v %w", dr.Error, res.Error)
-			}
-			// When both instant and range are true, we may overwrite the status code.
-			// To fix this (and other things) they should come in separate http requests.
-			dr.Status = res.Status
-		}
-		dr.Frames = append(dr.Frames, res.Frames...)
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			res := s.rangeQuery(traceCtx, client, q, enablePrometheusDataplane)
+			m.Lock()
+			addDataResponse(&res, dr)
+			m.Unlock()
+		}()
```

Comment:
Could you please move these changes and the exemplar changes behind the feature flag too? 
After talking to my squad we decided to keep all those changes behind the feature flag. That way it's going to be less risky.

Reply:
i have done the needful @itsmylife